packages/components/src/components/dialog/dialog.tsx

```diff
@@ -98,7 +98,7 @@ const Dialog = ({
         }
     };
 
-    const validateClickOutside = () => dismissable || (has_close_icon && is_visible && is_closed_on_cancel);
+    const validateClickOutside = () => !!dismissable || !!(has_close_icon && is_visible && is_closed_on_cancel);
```

Comment:
Function `validateClickOutside` is passed as an argument inside of `useOnClickOutside` hook. This hook is expecting function, which is returning `boolean`.